src/model/clients/HttpClient.ts

```diff
@@ -0,0 +1,276 @@
+import axios from 'axios';
+
+import { GenericResponse } from './GenericResponse';
+
+export enum HttpRequestMethods {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  PATCH = 'PATCH',
+  DELETE = 'DELETE',
+}
+
+export interface IHttpClientConfig {
+  timeout: number;
+  headers: Record<string, string>;
+  url: string;
+  method: HttpRequestMethods;
+  data?: Record<string, unknown>;
+  baseURL?: string;
+}
+
+export interface IHttpClient {
```

Comment:
Created an interface, so we could mock http clients in tests using plain objects